src/common/compress/src/decode.rs

```diff
@@ -409,9 +340,56 @@ impl<R: AsyncRead> AsyncRead for DecompressReader<R> {
     }
 }
 
+impl DecompressDecoder {
+    pub fn decompress_all(&mut self, compressed: &[u8]) -> common_exception::Result<Vec<u8>> {
+        let main = self.decompress_batch(compressed)?;
+        let tail = self.decompress_batch(&[])?;
+        if tail.is_empty() {
+            Ok(main)
+        } else {
+            let mut all = main;
+            all.extend_from_slice(&tail);
+            Ok(all)
+        }
+    }
+    // need to finish the decoding by adding a empty input
+    pub fn decompress_batch(&mut self, compressed: &[u8]) -> common_exception::Result<Vec<u8>> {
```

Comment:
why not expose the writer in this api:
```
decompress_batch(&mut self, compressed: &[u8],  buffer: &mut Vec<u8> )
```

Reply:
with a Guard to truncate the size if an error happens.

Reply:
buf is local, no need for guard?

Reply:
> why not expose the writer in this api:
> 
> ```
> decompress_batch(&mut self, compressed: &[u8],  buffer: &mut Vec<u8> )
> ```

decompress_all is only used for tests;
this pr was originally for add compress utils, when testing it, found the bug for decompress, so let`s fix it first.
we can lever the improvement for api and performance to another issue.